titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/DefaultKubeJobManagementReconciler.java

```diff
@@ -167,13 +205,15 @@ private void reconcile() {
         List<Task> tasks = v3JobOperations.getTasks();
 
         Map<String, V1Node> nodesById = nodes.stream().collect(Collectors.toMap(
-                node -> node.getMetadata().getName(),
+                node -> KubeUtil.getMetadataName(node.getMetadata()),
```

Comment:
should probably filter out entries with `null` (or empty strings) as their name since this is an index